internal/store/store.go

```diff
@@ -521,46 +601,83 @@ func (s *Store) DeleteCompose(id uuid.UUID) error {
 
 		var err error
 		if s.stateDir != nil {
-			err = os.RemoveAll(*s.stateDir + "/outputs/" + id.String())
+			// TODO: we need to rename the files as the compose will have multiple images
+			for _, imageBuild := range compose.ImageBuilds {
+				if imageBuild.QueueStatus == common.IBFinished || imageBuild.QueueStatus == common.IBFailed {
+					err = os.RemoveAll(s.getImageLocationForLocalTarget(id, imageBuild.ImageType))
+					if err != nil {
+						return err
+					}
+				}
+			}
 		}
 
 		return err
 	})
 }
 
-func (s *Store) PopCompose() Job {
+// PopJob returns a job from the job queue and changes the status of the corresponding image build to running
+func (s *Store) PopJob() Job {
 	job := <-s.pendingJobs
-	s.change(func() error {
+	// FIXME: handle or comment this possible error
+	_ = s.change(func() error {
+		// Get the compose from the map
 		compose, exists := s.Composes[job.ComposeID]
-		if !exists || compose.QueueStatus != "WAITING" {
+		// Check that it exists
+		if !exists {
 			panic("Invalid job in queue.")
 		}
-		compose.JobStarted = time.Now()
-		compose.QueueStatus = "RUNNING"
-		for _, t := range compose.Targets {
-			t.Status = "RUNNING"
+		// Loop over the image builds and find the right one according to the image type
+		// Change queue status to running for the image build as well as for the targets
+		for n, imageBuild := range compose.ImageBuilds {
+			imgTypeCompatStr, exists := imageBuild.ImageType.ToCompatString()
+			if !exists {
+				panic("fatal error, image type should exist but it does not")
+			}
+			if imgTypeCompatStr != job.ImageType {
+				continue
+			}
+			if imageBuild.QueueStatus != common.IBWaiting {
+				panic("Invalid job in queue.")
+			}
+			compose.ImageBuilds[n].QueueStatus = common.IBRunning
+			for m := range compose.ImageBuilds[n].Targets {
+				compose.ImageBuilds[n].Targets[m].Status = common.IBRunning
+			}
 		}
+		// Replace the compose struct with the new one
+		// TODO: I'm not sure this is needed, but I don't know what is the golang semantics in this case
 		s.Composes[job.ComposeID] = compose
 		return nil
 	})
 	return job
 }
 
-func (s *Store) UpdateCompose(composeID uuid.UUID, status string, image *Image, result *common.ComposeResult) error {
+// UpdateImageBuildInCompose sets the status and optionally also the final image.
+func (s *Store) UpdateImageBuildInCompose(composeID uuid.UUID, imageBuildID int, status common.ImageBuildState, image *compose.Image, result *common.ComposeResult) error {
 	return s.change(func() error {
-		compose, exists := s.Composes[composeID]
+		// Check that the compose exists
+		currentCompose, exists := s.Composes[composeID]
 		if !exists {
 			return &NotFoundError{"compose does not exist"}
 		}
-		if compose.QueueStatus == "WAITING" {
+		// Check that the image build was waiting
+		if currentCompose.ImageBuilds[imageBuildID].QueueStatus == common.IBWaiting {
 			return &NotPendingError{"compose has not been popped"}
 		}
 
```

Comment:
yeah, I missed that one, but I decided to revert the change because RCM API will not use local target and Weldr API will not use multiple image builds per compose, so it was useless anyway